### PR TITLE
Update ubuntu/mantic to 24.1

### PR DESCRIPTION
**Pull request by holmanb**
## Additional Context
This syncs ubuntu/mantic to 24.1 for the [SRU](https://bugs.launchpad.net/ubuntu/+source/cloud-init/+bug/2056100), will create a release branch for mantic after this lands. Will follow with Jammy and Focal once this is approved.

- manual patch refreshes
- synced changelogs from 23.4.x

In the github UI I see merge conflicts, yet testing locally I do not. I'm not sure if this is a temporary GitHub web issue or if I'm just missing something.

## Test Steps
```
quilt push -a
tox -e py3
quilt pop -a
```

## Merge type

- [ ] Squash merge using "Proposed Commit Message"
- [x] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

**Review by blackboxsw — CHANGES_REQUESTED** [review]
I'm seeing strange diffs against your quilt patches. I also didn't have to manually refresh any quilt patches as you mentioned.

When I look at the diff in this changset, it looks like something dropped the `return 2` -> `return 0`  quilt patch out of `d/patches/status-do-not-remove-duplicated-data.patch`.

I see the following diff against your branch (with my d/changelog comments applied)
```diff
diff --git a/debian/changelog b/debian/changelog
index c00883753..769be9260 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,4 +1,4 @@
-cloud-init (24.1.1-0ubuntu1~23.10.1) UNRELEASED; urgency=medium
+cloud-init (24.1.1-0ubuntu1~23.10.1) mantic; urgency=medium
 
   * d/apport-general-hook.py: Move apport hook to main branch
   * d/cloud-init.maintscript: remove /etc/cloud/clean.d/README
@@ -16,14 +16,14 @@ cloud-init (24.1.1-0ubuntu1~23.10.1) UNRELEASED; urgency=medium
     [0] https://github.com/canonical/cloud-init/pull/4799
     [1] commit/b519d861aff8b44a0610c176cb34adcbe28df144
   * refresh patches:
-    - d/p/revert-551f560d-cloud-config-after-snap-seeding.patch
     - d/p/status-do-not-remove-duplicated-data.patch
     - d/p/status-retain-recoverable-error-exit-code.patch
   * Upstream snapshot based on 24.1. (LP: #2056100).
     List of changes from upstream can be found at
     https://raw.githubusercontent.com/canonical/cloud-init/24.1/ChangeLog
 
- -- Brett Holman <brett.holman@canonical.com>  Mon, 04 Mar 2024 17:38:12 -0700
+ -- Chad Smith <chad.smith@canonical.com>  Mon, 04 Mar 2024 20:41:22 -0700
+
 
 cloud-init (23.4.4-0ubuntu0~23.10.1) mantic; urgency=medium
 
@@ -44,6 +44,7 @@ cloud-init (23.4.3-0ubuntu0~23.10.1) mantic; urgency=medium
 cloud-init (23.4.2-0ubuntu0~23.10.1) mantic; urgency=medium
 
   * Upstream snapshot based on 23.4.2. (LP: #2045582).
+    List of changes from upstream can be found at
     https://raw.githubusercontent.com/canonical/cloud-init/23.4.2/ChangeLog
     - Bugs fixed in this snapshot: (LP: #2051147)
 
diff --git a/debian/patches/status-do-not-remove-duplicated-data.patch b/debian/patches/status-do-not-remove-duplicated-data.patch
index cc11938b6..14bc47cd1 100644
--- a/debian/patches/status-do-not-remove-duplicated-data.patch
+++ b/debian/patches/status-do-not-remove-duplicated-data.patch
@@ -7,27 +7,19 @@ Last-Update: 2023-10-31
 This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
 --- a/cloudinit/cmd/status.py
 +++ b/cloudinit/cmd/status.py
-@@ -178,6 +178,8 @@
+@@ -178,6 +178,9 @@ def print_status(args, details: StatusDe
          "last_update": details.last_update,
          **details.v1,
      }
 +    details_dict["schemas"] = {"1": deepcopy(details_dict)}
 +    details_dict["_schema_version"] = "1"
++
      if args.format == "tabular":
          prefix = "\n" if args.wait else ""
  
-@@ -251,7 +253,7 @@
-         return 1
-     # Recoverable error
-     elif details.condition_status == ConditionStatus.DEGRADED:
--        return 2
-+        return 0
-     return 0
- 
- 
 --- a/tests/unittests/cmd/test_status.py
 +++ b/tests/unittests/cmd/test_status.py
-@@ -507,6 +507,7 @@
+@@ -507,6 +507,7 @@ PATH=/usr/local/sbin:/usr/local/bin:/usr
                  dedent(
                      """\
                     ---
@@ -35,7 +27,7 @@ This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
                     boot_status_code: enabled-by-kernel-cmdline
                     datasource: ''
                     detail: 'Running in stage: init'
-@@ -520,6 +521,23 @@
+@@ -520,6 +521,23 @@ PATH=/usr/local/sbin:/usr/local/bin:/usr
                         start: 123.45
                     last_update: Thu, 01 Jan 1970 00:02:04 +0000
                     recoverable_errors: {}
@@ -59,7 +51,7 @@ This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
                     stage: init
                     status: running
                     ...
-@@ -552,6 +570,25 @@
+@@ -552,6 +570,25 @@ PATH=/usr/local/sbin:/usr/local/bin:/usr
                      "init-local": {"finished": 123.46, "start": 123.45},
                      "last_update": "Thu, 01 Jan 1970 00:02:04 +0000",
                      "recoverable_errors": {},
@@ -85,7 +77,7 @@ This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
                      "stage": "init",
                  },
                  id="running_json_format",
-@@ -583,6 +620,7 @@
+@@ -583,6 +620,7 @@ PATH=/usr/local/sbin:/usr/local/bin:/usr
                  MyArgs(long=False, wait=False, format="json"),
                  1,
                  {
@@ -93,7 +85,7 @@ This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
                      "boot_status_code": "enabled-by-kernel-cmdline",
                      "datasource": "nocloud",
                      "detail": (
-@@ -604,6 +642,32 @@
+@@ -604,6 +642,30 @@ PATH=/usr/local/sbin:/usr/local/bin:/usr
                      },
                      "last_update": "Thu, 01 Jan 1970 00:02:05 +0000",
                      "recoverable_errors": {},
@@ -121,12 +113,10 @@ This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
 +                            "status": "error",
 +                        }
 +                    },
-+                    "last_update": "Thu, 01 Jan 1970 00:02:05 +0000",
-+                    "recoverable_errors": {},
                      "stage": None,
                  },
                  id="running_json_format_with_errors",
-@@ -666,6 +730,7 @@
+@@ -666,6 +728,7 @@ PATH=/usr/local/sbin:/usr/local/bin:/usr
                  MyArgs(long=False, wait=False, format="json"),
                  2,
                  {
@@ -134,7 +124,7 @@ This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
                      "boot_status_code": "enabled-by-kernel-cmdline",
                      "datasource": "nocloud",
                      "detail": (
-@@ -725,6 +790,89 @@
+@@ -725,6 +788,89 @@ PATH=/usr/local/sbin:/usr/local/bin:/usr
                              "don't try to open the hatch or we'll all be soup"
                          ],
                      },
diff --git a/debian/patches/status-retain-recoverable-error-exit-code.patch b/debian/patches/status-retain-recoverable-error-exit-code.patch
index a6894b8e2..d5d839060 100644
--- a/debian/patches/status-retain-recoverable-error-exit-code.patch
+++ b/debian/patches/status-retain-recoverable-error-exit-code.patch
@@ -4,9 +4,20 @@ Author: Alberto Contreras <alberto.contreras@canonical.com>
 Last-Update: 2024-01-08
 ---
 This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
+--- a/cloudinit/cmd/status.py
++++ b/cloudinit/cmd/status.py
+@@ -254,7 +254,7 @@ def handle_status_args(name, args) -> in
+         return 1
+     # Recoverable error
+     elif details.condition_status == ConditionStatus.DEGRADED:
+-        return 2
++        return 0
+     return 0
+ 
+ 
 --- a/tests/unittests/cmd/test_status.py
 +++ b/tests/unittests/cmd/test_status.py
-@@ -728,7 +728,7 @@
+@@ -726,7 +726,7 @@ PATH=/usr/local/sbin:/usr/local/bin:/usr
                  },
                  None,
                  MyArgs(long=False, wait=False, format="json"),
```

**Comment by holmanb** [comment]
> I'm seeing strange diffs against your quilt patches. I also didn't have to manually refresh any quilt patches as you mentioned.

I just noticed that the docs I was following were incorrect and were syncing origin rather than upstream. I just updated the docs and then followed them - and took your requested changelog changes into account.

@blackboxsw ready for re-review

File: debian/changelog [hunk 1]
```diff
@@ -1,4 +1,4 @@
-cloud-init (23.4.1-0ubuntu1~23.10.3) UNRELEASED; urgency=medium
+cloud-init (24.1.1-0ubuntu1~23.10.1) mantic; urgency=medium
```
Comment (blackboxsw) [inline debian/changelog:1]: Looks like our [tooling doesn't yet handle a shift from major.minor.hotfix in d/changelog to a new commit based on just major.minor](https://github.com/canonical/uss-tableflip/pull/119). We need to manually drop this patch version as this is our first release of 24.1 to mantic and our tooling wasn't aware of the shift to the next major.minor so it left the trailing patch in place.
```suggestion
cloud-init (24.1-0ubuntu1~23.10.1) mantic; urgency=medium
```


```suggestion
cloud-init (24.1-0ubuntu1~23.10.1) mantic; urgency=medium
```

**Review by blackboxsw — CHANGES_REQUESTED** [review]
Thanks @holmanb one more thing I overlooked on the first pass.

**Comment by holmanb** [comment]
> Thanks @holmanb one more thing I overlooked on the first pass.

Fixed - also the commit message too

**Comment by blackboxsw** [comment]
This PR is really waiting on closeout of the known 24.1.1 issues we want to sync. I'm recusing myself from review here as I am leaving on PTO for a bit. Once we have landed all critical content for 24.1.1 release we'll need to revisit this SRU to mantic and ensure that this SRU to mantic includes the 24.1.1 bug-fix release that will first have to land in noble with fixes for Oracle, Oracle network schema support of broadcase,  VMware AttributeError fixes and ubuntu_advantage -> pro renames.

**Pull request closed without merging**